JASP-Engine/JASP/R/principalcomponentanalysis.R

```diff
@@ -605,54 +662,51 @@ PrincipalComponentAnalysis <- function(dataset = NULL, options, perform = "run",
 
 
 # Factor correlations:
-.getFactorCorrelationsPCA <- function(analysisResults, options,perform){
-  
-  # Extract loadings:
-  if (!is.null(analysisResults)){
-    
-    corMatrix <- as.matrix(analysisResults$r.scores)
-    # loadingsMatrix <- matrix(unlist(loadingsMatrix),nrow(loadingsMatrix),ncol(loadingsMatrix))
-    
-    
-    # loadingsMatrix <- matrix(1,12,1)
-    
-  } else {
-    
+.getFactorCorrelations <- function(analysisResults, options, perform, type){
+
+  # Create JASP table:
+  FactorCorrelations <- list()
+  FactorCorrelations[["title"]] <- ifelse(type == "pca","Component Correlations", "Factor Correlations")
+
+   # Extract loadings  
+  if(is.null(analysisResults) || isTryError(analysisResults)){#|| perform == "init"
     if (is.null(options$numberOfFactors)){
       nFact <- 0
     } else {
       nFact <- options$numberOfFactors
     }
     
     corMatrix <- matrix(,0,0)
-    
+
+
+    if(isTryError(analysisResults)){
+      footnotes <- .newFootnotes()
+      errorMessage <- .extractErrorMessage(analysisResults)
+      .addFootnote(footnotes, symbol="<em>Error.</em>", text=errorMessage)
```

Comment:
@TimKDJ error message?

Reply:
error message removed, not sure how to present it properly